ui/apps/platform/cypress/integration/vulnmanagement/reporting/reportingForm.test.js

```diff
@@ -123,4 +130,112 @@ describe('Vulnerability Management Reporting form', () => {
+
+        // Delete collection from previous calls, if present
+        tryDeleteCollection(collectionName);
+        tryDeleteIntegration(integrationsSource, emailNotifierName);
+        tryDeleteVMReportConfigs(reportName);
+
+        visitVulnerabilityReportingToCreate({}, isCollectionsEnabled);
+
+        cy.get('h1:contains("Create an image vulnerability report")');
+
+        // Fill out report details
+        getInputByLabel('Report name').type(reportName);
+        getInputByLabel('Description').type(reportDescription);
+
+        getInputByLabel('Repeat report…').click();
+        cy.get('button[role="option"]:contains("Monthly")').click();
+
+        cy.get('button:has(*:contains("Select days"))').click();
+        cy.get('*[role="listbox"] span:contains("The middle of the month")').click();
+
+        getInputByLabel('Distribution list').type('scooby@mysteryinc.com');
+
+        cy.get('button:has(*:contains("Fixable states selected"))').click();
+        cy.get('*[role="listbox"] span:contains("Unfixable")').click();
+
+        cy.get('button:has(*:contains("Severities selected"))').click();
+        cy.get('*[role="listbox"] span:contains("Moderate")').click();
+        cy.get('*[role="listbox"] span:contains("Low")').click();
+
+        // Create an email notifier via modal
+        cy.get('button:contains("Create email notifier")').click();
+        getInputByLabel('Integration name').type(emailNotifierName);
+        getInputByLabel('Email server').type('e2e.test.rox.systems:465');
+        getInputByLabel('Enable unauthenticated SMTP').click();
+        getInputByLabel('Sender').type('e2e-test-sender@rox.systems');
+        getInputByLabel('Default recipient').type('e2e-test-recipient@rox.systems');
+        cy.get('*[role="dialog"] button:contains("Save integration")').click();
+
+        // The newly created notifier should automatically be selected in the input
+        cy.get(`button[aria-label="Select a notifier"]:contains("${emailNotifierName}")`);
+
+        // Create a collection via modal
+        cy.get('button:contains("Create collection")').click();
+
+        cy.get('*[role="dialog"] input[name="name"]').type(collectionName);
+        cy.get('*[role="dialog"] button:contains("All namespaces")').click();
+        cy.get('*[role="dialog"] button:contains("Namespaces with names matching")').click();
+        cy.get(
+            '*[role="dialog"] input[aria-label="Select value 1 of 1 for the namespace name"]'
+        ).type('stackrox');
+        cy.get(`*[role="dialog"] button:contains('stackrox')`).click();
+        cy.get('*[role="dialog"] button:contains("Save")').click();
+
+        // The newly created collection should automatically be selected
+        cy.get(`input[placeholder="Select a collection"]`)
+            .invoke('val')
+            .should('equal', collectionName);
+
+        // Create the VM Report config
+        cy.get('button[data-testid="create-btn"]').click();
```

Comment:
Low priority: can we write `:contains` pseudo-selector and delete `data-testid` attribute in element?

Reply:
Unfortunately in this case we can't, since the `button:contains("Create")` selector will match both "Create collection" and "Create email notifier" in the page. I used `data-testid` since it was already present, but we could probably scope the selector by relying on the DOM structure more as a follow up.